srcts/src/components/webComponents.ts

```diff
@@ -1,8 +1,15 @@
 import { BslibTooltip } from "./tooltip";
 import { BslibPopover } from "./popover";
+import { DarkModeSwitch } from "./darkModeSwitch";
 import { makeInputBinding } from "./webcomponents/_makeInputBinding";
+import { shinyAddCustomMessageHandlers } from "./webcomponents/_shinyAddCustomMessageHandlers";
 
-[BslibTooltip, BslibPopover].forEach((cls) => {
+[BslibTooltip, BslibPopover, DarkModeSwitch].forEach((cls) => {
   customElements.define(cls.tagName, cls);
-  if (cls.isShinyInput) makeInputBinding(cls.tagName);
+  if (window.Shiny) {
+    if (cls.isShinyInput) makeInputBinding(cls.tagName);
+    if ("shinyCustomMessageHandlers" in cls) {
+      shinyAddCustomMessageHandlers(cls["shinyCustomMessageHandlers"]);
+    }
+  }
```

Comment:
This outer `window.Shiny` isn't required, since the check also happens inside both `makeInputBinding` and `shinyAddCustomMessageHandlers`. But I like that it's explicit and shows clearly from here that these additional steps only happen in a shiny context.